src/Microsoft.DncEng.Configuration.Extensions/KeyVaultConfigMapper.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Text.RegularExpressions;
+using Azure;
+using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.DncEng.Configuration.Extensions
+{
+    public static class KeyVaultConfigMapper
+    {
+        private static readonly Regex VaultReferenceRegex = new Regex(@"\[vault\((?<key>[^])]+)\)\]");
+
+        public static Func<string, string> Create(IConfiguration bootstrapConfiguration)
+        {
+            var keyVaultUri = bootstrapConfiguration[ConfigurationConstants.KeyVaultUriConfigurationKey];
+            var credentials = ServiceConfigurationExtensions.GetAzureTokenCredential(bootstrapConfiguration);
+            var client = new SecretClient(new Uri(keyVaultUri), credentials);
```

Comment:
Can we save this client? This is going to run for a lot of values, isn't it?

Reply:
It is saved because this method is called once per call to `services.AddDefaultJsonConfiguration`. The client is saved in the closure allocated for the returned method.

Reply:
Oooo, yeah, this just returns a func, I didn't realize that. :-)